cmd/tunnel/tunnel.go

```diff
@@ -106,10 +120,16 @@ func main() {
 	}
 }
 
-func tlsConfig(cert tls.Certificate, c *ClientConfig) *tls.Config {
+func tlsConfig(cert tls.Certificate, rootca *x509.CertPool, c *ClientConfig) *tls.Config {
```

Comment:
Could you change the signature of this function to `func tlsConfig(config *ClientConfig) (*tls.Config, error)` and move all the tls related processing there.

Reply:
Also it might be good to rename `c` to `config` in the main function.